### PR TITLE
Alex/calendar

### PR DESCRIPTION
**Pull request by alexyoon97**
### What this PR does (required):
- Displaying user's events on the calendar
- drag and drop functionality implemented
- show more buttons available for popup dialogue when the event overloads

### Screenshots / Videos (required):
calendar_1
calendar_2

### Any information needed to test this feature (required):
- Sign up as a user and make events and see the events on the calendar view.

### Any issues with the current functionality (optional):
-Not many issues but can not find right types for
withDragAndDrop(Calendar as *any),
const moveEvent = ({ event, start, end }: { event: *any; start: *any; end: *any })
suggestions?

### CODE REVIEW

File: client/src/App.tsx [hunk 1]
```diff
@@ -18,14 +19,17 @@ function App(): JSX.Element {
           <AuthProvider>
             <SocketProvider>
               <Switch>
-                <Route exact path="/login" component={Login} />
+                <Route exact path="/login" component={Calendar} />
```
Comment (ajaytomgeorge) [inline client/src/App.tsx]: May i know why are you routing login path to Calendar component?, Copy paste error?

File: client/src/components/Calendar/DnDCaldendar.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import { Calendar, momentLocalizer } from 'react-big-calendar';
+import 'react-big-calendar/lib/css/react-big-calendar.css';
+import withDragAndDrop from 'react-big-calendar/lib/addons/dragAndDrop';
+import moment from 'moment';
+import { IUserSchedule, mockDatas } from './MockEvent';
+import { useState } from 'react';
```
Comment (ajaytomgeorge) [inline client/src/components/Calendar/DnDCaldendar.tsx]: You can sort these imports

There are 3 main sections of import or require statements. These are separated by a blank line.
1. Optionally, anything required to make the whole module/app work, like config.
2. 3rd-party code, usually in the form of named npm packages.
3. App-level code – in other words any local imports.

Also can you try this 
import React, { useState } from "react";

File: client/src/components/Calendar/DnDCaldendar.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+import './styles.css';
+import { Toolbar, EventComponent } from './extension';
+import { FetchOptions } from '../../interface/FetchOptions';
+
+// import { useAuth } from '../../context/useAuthContext';
+// import { useSocket } from '../../context/useSocketContext';
+// import { useHistory } from 'react-router-dom';
```
Comment (ajaytomgeorge) [inline client/src/components/Calendar/DnDCaldendar.tsx]: Remove theses comments (Reserve comments to cases that needs understanding by future developers.)

File: client/src/components/Calendar/DnDCaldendar.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  // const { initSocket } = useSocket();
+  // const history = useHistory();
+
+  // useEffect(() => {
+  //   initSocket();
+  // }, [initSocket]);
+
+  // if (loggedInUser === undefined) return <CircularProgress />;
+  // if (!loggedInUser) {
+  //   history.push('/login');
```
Comment (ajaytomgeorge) [inline client/src/components/Calendar/DnDCaldendar.tsx]: Same as above

File: client/src/components/Calendar/DnDCaldendar.tsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  //   // loading for a split seconds until history.push works
+  //   return <CircularProgress />;
+  // }
+
+  const moveEvent = ({ event, start, end }: { event: any; start: any; end: any }): void => {
```
Comment (ajaytomgeorge) [inline client/src/components/Calendar/DnDCaldendar.tsx]: Can you make this more readable, something like

```
const moveEvent = ({ event, start, end }: MEvent): void => {
```

File: client/src/components/Calendar/DnDCaldendar.tsx [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+        error: { message: 'Unable to connect to server. Please try again' },
+      }));
+  };
+  return (
+    <DragAndDropCalendar
+      selectable
+      timeslots={2}
+      localizer={localizer}
+      events={events}
+      onDragStart={console.log}
```
Comment (ajaytomgeorge) [inline client/src/components/Calendar/DnDCaldendar.tsx]: May I know why you are passing, console.log?

File: client/src/components/Calendar/extension.tsx [hunk 7]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { ToolbarProps, Event, EventProps } from 'react-big-calendar';
+import { CalendarEvent, CalendarResource } from './interface';
+import { Typography } from '@material-ui/core';
+
+export class Toolbar extends React.Component<ToolbarProps<CalendarEvent, CalendarResource>> {
```
Comment (ajaytomgeorge) [inline client/src/components/Calendar/extension.tsx]: Why use class components over functional components?

File: client/src/components/Calendar/extension.tsx [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+      <div className="rbc-toolbar">
+        <Typography variant="h4" className="rbc-toolbar-label">
+          {label}
+        </Typography>
+      </div>
+    );
+  }
+}
+
+export class EventComponent extends React.Component<EventProps<Event>> {
```
Comment (ajaytomgeorge) [inline client/src/components/Calendar/extension.tsx]: same ques as above..

**Review by ajaytomgeorge — COMMENTED** [review]
@alexyoon97  Good work, :-) some comments though

File: client/src/components/Calendar/extension.tsx (hunk 8 above)
Comment (alexyoon97) [reply client/src/components/Calendar/extension.tsx]: Okay! thanks for your comments ajay, I will fix my code as soon as possible

File: client/src/App.tsx (hunk 1 above)
Comment (alexyoon97) [reply client/src/App.tsx]: I swap two components just to render the calendar easy way, let me know if there's a better way!

Comment (alexyoon97) [reply client/src/App.tsx]: I changed back before I commit, but it did not update properly

File: client/src/components/Calendar/DnDCaldendar.tsx (hunk 3 above)
Comment (alexyoon97) [reply client/src/components/Calendar/DnDCaldendar.tsx]: Sure! I will remove my comments

File: client/src/components/Calendar/DnDCaldendar.tsx (hunk 5 above)
Comment (alexyoon97) [reply client/src/components/Calendar/DnDCaldendar.tsx]: Sounds good, thanks.

File: client/src/components/Calendar/DnDCaldendar.tsx (hunk 6 above)
Comment (alexyoon97) [reply client/src/components/Calendar/DnDCaldendar.tsx]: Checking the event status, it can be removed.

**Review by moffatethan — APPROVED** [review]

**Pull request merged**